### PR TITLE
Run-time NDIS version, RSS UDP and device RSS support

### PR DESCRIPTION
**Pull request by ybendito**
1. Support NDIS version higher than 6.30 in Win10 build. The driver reports 6.80 when the OS has NDIS 6.80 or more, otherwise reports 6.30 (this can be easily changed to support also intermediate NDIS versions but I do not see enough reasons for that)
2. Support RSS UDP for run-time 6.80 and up
3. Support device RSS features and hash delivery. WE STILL CALCULATE the hash and keep internal statistics about hits misses (i.e. when the value reported by the device is the same/different from one calculated by the driver)

### CODE REVIEW

File: NetKVM/wlh/ParaNdis6-Oid.cpp [hunk 1]
```diff
@@ -1243,7 +1243,11 @@ void ParaNdis6_Fill620PowerCapabilities(PNDIS_PM_CAPABILITIES pPower620Caps)
     pPower620Caps->Header.Type = NDIS_OBJECT_TYPE_DEFAULT;
     pPower620Caps->Header.Revision = NDIS_PM_CAPABILITIES_REVISION_1;
     pPower620Caps->Header.Size = NDIS_SIZEOF_NDIS_PM_CAPABILITIES_REVISION_1;
-
+#if NDIS_SUPPORT_NDIS650
```
Comment (YanVugenfirer) [inline NetKVM/wlh/ParaNdis6-Oid.cpp:1289]: Shouldn't we use here also "if (CheckNdisVersion())" construct here?
The function itself is under 6.2 define and revision 2 supported form 6.3.

Comment (YanVugenfirer) [reply NetKVM/wlh/ParaNdis6-Oid.cpp:1289]: Same as here: https://github.com/virtio-win/kvm-guest-drivers-windows/pull/450/commits/3b32c43e31eebfd88cf213d05121fdf85d73e35b

**Comment by YanVugenfirer** [comment]
HCK failure in "NDISTest 6.0 - [2 Machine] - 2c_OidsNdisRequest":

For some reason the test doesn't like the new vustom OID: 0xff010203

>>>>
Failed: |  
-- | --
Variation #228 | [88888] Able to set unrecognized OID 0xFF010203

**Comment by YanVugenfirer** [comment]
"Failed: NDISTest 6.5 - [1 Machine] - StandardizedKeywords" failed as well.

Failed (2)TestCaseIDAssertionIDDescription0000000000*RSCIPv4 NDI\Params\Default is not "0" as expected!0000000000*RSCIPv6 NDI\Params\Default is not "0" as expected! | Failed (2)TestCaseIDAssertionIDDescription0000000000*RSCIPv4 NDI\Params\Default is not "0" as expected!0000000000*RSCIPv6 NDI\Params\Default is not "0" as expected! | Failed (2) | TestCaseID | AssertionID | Description | 00000 | 00000 | *RSCIPv4 NDI\Params\Default is not "0" as expected! | 00000 | 00000 | *RSCIPv6 NDI\Params\Default is not "0" as expected!
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
Failed (2)TestCaseIDAssertionIDDescription0000000000*RSCIPv4 NDI\Params\Default is not "0" as expected!0000000000*RSCIPv6 NDI\Params\Default is not "0" as expected! | Failed (2) | TestCaseID | AssertionID | Description | 00000 | 00000 | *RSCIPv4 NDI\Params\Default is not "0" as expected! | 00000 | 00000 | *RSCIPv6 NDI\Params\Default is not "0" as expected!
Failed (2)
TestCaseID | AssertionID | Description
00000 | 00000 | *RSCIPv4 NDI\Params\Default is not "0" as expected!
00000 | 00000 | *RSCIPv6 NDI\Params\Default is not "0" as expected!

File: NetKVM/wlh/ParaNdis6-Oid.cpp (hunk 1 above)
Comment (ybendito) [reply NetKVM/wlh/ParaNdis6-Oid.cpp:1289]: Run-time NDIS version currently is:
6.30 for any system with NDIS < 6.80
6.80 for any system with NDIS >= 6.80
So CheckNdisVersion currently allows to distinguish between 6.80 and 6.30.
For <= Win 8.1 the code here stays fully backward compatible (intentionally).
For all the Win 10 we indicate revision 2

**Comment by YanVugenfirer** [comment]
For Q35 OID test failed as well (expected). By the way, maybe a good time to check the warnings as well:

Failed: |  
-- | --
Variation #228 | [88888] Able to set unrecognized OID 0xFF010203

**Comment by YanVugenfirer** [comment]
Warned: |  
-- | --
Variation #5 | Unrecognized nonprivate OID (0x0002021d) value detected!
Variation #6 | Unrecognized nonprivate OID (0x0002021d) value detected!
Variation #10 | Unrecognized nonprivate OID (0x0002021d) value detected!
Variation #28 | Version number mismatch between file and OID
Variation #67 | Unrecognized nonprivate OID (0x0002021d) value detected!
Variation #171 | Test loop ran too long (50 times). Moving on

File: NetKVM/wlh/ParaNdis6-Oid.cpp (hunk 1 above)
Comment (YanVugenfirer) [reply NetKVM/wlh/ParaNdis6-Oid.cpp:1289]: Maybe worth adding a comment.

**Comment by ybendito** [comment]
Fixed HCK failure in OID test
Changed mistake in PhysicalMediaType read
The "Keywords" test must be filtered, there is a filter for RSC keyword

**Comment by YanVugenfirer** [comment]
1. SDV will fail with this PR.
2. Next PR will fix this.

**Pull request merged**